File: 67-add-binary/add-binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
function addBinary(a, b) {
    let result = '';
    let carry = 0;
    let i = a.length - 1;
    let j = b.length - 1;

    while (i >= 0 || j >= 0 || carry) {
        let sum = carry;
        if (i >= 0) {
            sum += Number(a[i--]); // Convert char to int
        }
        if (j >= 0) {
            sum += Number(b[j--]); // Convert char to int
        }
        result = (sum % 2) + result; // Prepend the binary result
        carry = Math.floor(sum / 2); // Calculate carry
    }

    return result;
}

File: 67-add-binary/add-binary.cpp
#include <iostream>
#include <string>
#include <algorithm>

class Solution {
public:
    std::string addBinary(std::string a, std::string b) {
        std::string result;
        int carry = 0;
        int i = a.size() - 1;
        int j = b.size() - 1;

        while (i >= 0 || j >= 0 || carry) {
            int sum = carry;
            if (i >= 0) {
                sum += a[i--] - '0'; // Convert char to int
            }
            if (j >= 0) {
                sum += b[j--] - '0'; // Convert char to int
            }
            result.push_back((sum % 2) + '0'); // Append the binary result
            carry = sum / 2; // Calculate carry
        }

        std::reverse(result.begin(), result.end()); // Reverse the result
        return result;
    }
};